source/implementations/f7/Meadow.F7/Devices/F7CellNetworkAdapter.cs

```diff
@@ -206,5 +212,48 @@ public CellNetwork[] Scan()
     {
         return Core.Interop.Nuttx.MeadowCellNetworkScanner();
     }
+    
+    /// <summary>
+    /// Returns error according to an input value, otherwise <b>Undefined Cell error</b>
+    /// </summary>
+    private string ParseError(string input)
+    {
+        if (input !=null)
+        {
+            string pattern = @"\+CME ERROR: (.+)$";
+            Match match = Regex.Match(input, pattern);
 
+            if (match.Success && match.Groups.Count >= 2)
+            {
+                return match.Groups[1].Value;
+            }
+        }
+        return "Undefined Cell error";
+    }
+
+    /// <summary>
+    /// Returns the error string, otherwise <b>Undefined Cell error</b>
+    /// </summary>
+    private string GetError()
+    {
+        int errno = Core.Interop.Nuttx.meadow_get_cell_error();
+        
+        CellError cellError= (CellError)errno;
+
+        switch(cellError)
```

Comment:
I think that we can help the users troubleshoot by giving a bit more verbose errors:
```
Error: Invalid cellular network settings. Please check your cell configuration file."
Error: The selected cell module is invalid. Please ensure you have configured the right module name.
```